src/components/organization/configuration-settings/CreateDid.tsx

```diff
@@ -1,121 +1,147 @@
-		try {
-			const { data } = (await getLedgerConfig()) as AxiosResponse;
-			if (data?.statusCode === apiStatusCodes.API_STATUS_SUCCESS) {
-				const ledgerdata: ILedgerConfig = {};
-				data.data.forEach((item: ILedgerItem) => {
-					ledgerdata[item.name.toLowerCase()] = { ...item.details };
-				});
-				setMappedData(ledgerdata);
-			}
-		} catch (err) {
-			console.error('Error in fetching ledger config:::', err);
-		}
-	};
+	const [ledgerValue, setLedgerValue] = useState<string | null>(null);
+	const [method, setMethod] = useState<string | null>(null);
+	const [networkValue, setNetworkValue] = useState<string | null>(null);
+	const [completeDidMethodValue, setCompleteDidMethodValue] = useState<string | null>(null);
+	const [havePrivateKey, setHavePrivateKey] = useState(false);
+	const [privateKeyValue, setPrivateKeyValue] = useState<string>('');
+	const [walletErrorMessage, setWalletErrorMessage] = useState<string | null>(null);
 
 	const fetchOrganizationDetails = async () => {
 		const orgId = await getFromLocalStorage(storageKeys.ORG_ID);
-		const response = await getOrganizationById(orgId as string);
+		const response = await getOrganizationById(orgId);
 		const { data } = response as AxiosResponse;
 		setLoading(false);
 		if (data?.statusCode === apiStatusCodes.API_STATUS_SUCCESS) {
 			const didMethod = data?.data?.org_agents[0]?.orgDid
 				?.split(':')
 				.slice(0, 2)
 				.join(':');
-			let getLedgerName;
-			switch (didMethod) {
-				case 'did:indy':
-				case 'did:polygon':
-					getLedgerName = data?.data?.org_agents[0]?.ledgers?.name;
-					setLedgerName(getLedgerName);
-
-					break;
-				case 'did:web':
-				case 'did:key':
-					getLedgerName = data?.data?.org_agents[0]?.orgDid
-						?.split(':')
-						.slice(1)[0];
-					setLedgerName(getLedgerName);
-
-					break;
-				default:
-					console.error('Unsupported DID format');
+			setMethod(didMethod);
+
+
+			let ledgerName;
+			if (didMethod === DidMethod.INDY || DidMethod.POLYGON) {
+				ledgerName = data?.data?.org_agents[0]?.orgDid.split(':')[1];
+			} else {
+				ledgerName = 'No Ledger';
+
 			}
+			setLedgerValue(ledgerName);
+
+			let networkName;
+
+			if (didMethod === DidMethod.INDY) {
+				networkName = data?.data?.org_agents[0]?.orgDid.split(':').slice(2, 4).join(':');
+			} else if (didMethod === DidMethod.POLYGON) {
+				networkName = data?.data?.org_agents[0]?.orgDid.split(':')[2];
+			} else {
+				networkName = '';
+			}
+			setNetworkValue(networkName);
+
+
+			let completeDidMethod;
+
+			if (didMethod === DidMethod.INDY) {
+				completeDidMethod = data?.data?.org_agents[0]?.orgDid.split(':').slice(0, 4).join(':');
+			} else {
+				completeDidMethod = didMethod;
+			}
+			setCompleteDidMethodValue(completeDidMethod);
+
 		} else {
 			console.error('Error in fetching organization:::');
 		}
-		setLoading(false);
 	};
 
 	useEffect(() => {
 		fetchOrganizationDetails();
 	}, []);
 
+
+	const checkBalance = async (privateKey: string) => {
```

Comment:
This function should also handle the mainnet flow

Reply:
Resolved